libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1650,3 +1671,216 @@ Type const& IRGeneratorForStatements::type(Expression const& _expression)
 	solAssert(_expression.annotation().type, "Type of expression not set.");
 	return *_expression.annotation().type;
 }
+
+bool IRGeneratorForStatements::visit(TryStatement const& _tryStatement)
+{
+	optional<ReturnInfo> outerReturnInfo(std::move(m_returnInfo));
+	ScopeGuard _leaveGuard = ScopeGuard([&]() { m_returnInfo.swap(outerReturnInfo); });
+
+	m_returnInfo.reset();
+	_tryStatement.externalCall().accept(*this);
+	solAssert(m_returnInfo.has_value(), "");
+
+	auto const trySuccessCondition = m_context.trySuccessConditionVariable(_tryStatement.externalCall());
+
+	m_code << "switch iszero(" << trySuccessCondition << ")\n";
+
+	m_code << "case 0 { // success case\n";
+	TryCatchClause const& successClause = *_tryStatement.clauses().front();
+	decodeTryCallReturnParameters(successClause, _tryStatement.externalCall().annotation().type);
```

Comment:
After this line ``m_returnInfo`` is not needed anymore. So I'd add a ``m_returnInfo.reset()`` right after this, remove the whole scope guard and assert in ``appendExternalFunctionCall`` that ``!m_returnInfo`` before calling ``m_returnInfo.emplace(...)``. Do you think that would work?